File: problems/0026-remove-duplicates-from-sorted-array/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let i = 0;
    let prev = 0;

    do {
        const n = nums[i];
        if (n === undefined) break;

        if (i > 0 && prev === n) {
            nums.splice(i, 1);
        } else {
            prev = n;
            ++i;
        }
    } while(1)
    
    return i; 
};

File: problems/0026-remove-duplicates-from-sorted-array/solution.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        ans = 0
        for i in range(1, len(nums)):
            if nums[ans] != nums[i]:
                ans += 1
                nums[ans] = nums[i]
        
        return ans + 1
